### PR TITLE
Add model card for MobileViT

### PR DESCRIPTION
**Pull request by Shivamjan**
# What does this PR do?

This PR adds a detailed and beginner-friendly model card for MobileViT to the Hugging Face Transformers documentation. The previous model card was minimal and lacked clear explanations about the model architecture. This model retains several elements from the earlier version, as they remain applicable and effective for users.

The new version includes:
- A clear explanation of the MobileViT architecture.
- Notes on preprocessing and image format.
- Clarifies how to use the model for classification and segmentation.
- Highlights TensorFlow Lite compatibility for mobile use.
- Primary references to the original paper and related resources.

Fixes # (issue)

## Before submitting
- [ x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by Shivamjan** [comment]
@stevhliu Please take a look at your convenience and do let me know if there is any further changes required.

File: docs/source/en/model_doc/mobilevit.md [hunk 1]
```diff
@@ -1,141 +1,214 @@
...
 
+
+
 # MobileViT
 
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-<img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
+
+<div style="float: left;">
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]:
```suggestion
<div style="float: right;">
```

File: docs/source/en/model_doc/mobilevit.md [hunk 2]
```diff
@@ -1,141 +1,214 @@
...
+
+**What makes MobileViT lightweight?** 
+Most of the models that uses Transformers for vision would first divide the images into several patches which are further flattened and converted into vectors. This causes in the loss of structural properties of an image, which isn't the case for CNNs. Now, this causes the Transformer models to go bigger and deeper to learn visual representations. 
+But MobileViT uses both convolutions and transformers in a way that the resultant block has convolution-like properties while simultaneously allowing for global interactions. This allows us to design a more shallow and narrow models, which are light-weight.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]: Aim to keep it nice and short

```suggestion
[MobileViT](https://huggingface.co/papers/2110.02178) is a lightweight vision transformer for mobile devices that merges CNNs's efficiency and inductive biases with transformers global context modeling. It treats transformers as convolutions, enabling global information processing without the heavy computational cost of standard ViTs.
```

File: docs/source/en/model_doc/mobilevit.md [hunk 3]
```diff
@@ -1,141 +1,214 @@
...
+
+![enter image description here](https://user-images.githubusercontent.com/67839539/136470152-2573529e-1a24-4494-821d-70eb4647a51d.png)
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]: Can you upload the image [here](https://huggingface.co/datasets/huggingface/documentation-images/tree/main/transformers/model_doc) and then ping me to merge please :)

The image should be formatted like:

```md
<div class="flex justify-center">
   <img>
</div>
```

File: docs/source/en/model_doc/mobilevit.md [hunk 4]
```diff
@@ -1,141 +1,214 @@
...
 
-The MobileViT model was proposed in [MobileViT: Light-weight, General-purpose, and Mobile-friendly Vision Transformer](https://huggingface.co/papers/2110.02178) by Sachin Mehta and Mohammad Rastegari. MobileViT introduces a new layer that replaces local processing in convolutions with global processing using transformers.
 
-The abstract from the paper is the following:
+You can find all the original MobileViT checkpoints under the [Apple](https://huggingface.co/apple) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]:
```suggestion
You can find all the original MobileViT checkpoints under the [Apple](https://huggingface.co/apple/models?search=mobilevit) organization.
```

File: docs/source/en/model_doc/mobilevit.md [hunk 5]
```diff
@@ -1,141 +1,214 @@
...
 
-This model was contributed by [matthijs](https://huggingface.co/Matthijs). The TensorFlow version of the model was contributed by [sayakpaul](https://huggingface.co/sayakpaul). The original code and weights can be found [here](https://github.com/apple/ml-cvnets).
+> [!TIP]
+> - This model was contributed by [Matthijs](https://huggingface.co/Matthijs).
+> - Use [`MobileViTImageProcessor`](https://huggingface.co/docs/transformers/main/en/model_doc/mobilevit#transformers.MobileViTImageProcessor) to preprocess images.
+> - If using custom preprocessing, ensure that images are in **BGR** format (not RGB), as expected by the pretrained weights.
+> - The **classification models** are pretrained on [**ImageNet-1k**](https://huggingface.co/datasets/imagenet-1k) (ILSVRC 2012).
+> - The **segmentation models** use a [**DeepLabV3**](https://huggingface.co/papers/1706.05587) head and are pretrained on [**PASCAL VOC**](http://host.robots.ox.ac.uk/pascal/VOC/).
+> - TensorFlow versions are compatible with **TensorFlow Lite**, making them ideal for edge/mobile deployment.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]:
```suggestion
> - This model was contributed by [matthijs](https://huggingface.co/Matthijs) and the TensorFlow version was contributed by [sayakpaul](https://huggingface.co/sayakpaul).
>
> Click on the MobileViT models in the right sidebar for more examples of how to apply MobileViT to different vision tasks.
```

File: docs/source/en/model_doc/mobilevit.md [hunk 6]
```diff
@@ -1,141 +1,214 @@
...
-- One can use [`MobileViTImageProcessor`] to prepare images for the model. Note that if you do your own preprocessing, the pretrained checkpoints expect images to be in BGR pixel order (not RGB).
-- The available image classification checkpoints are pre-trained on [ImageNet-1k](https://huggingface.co/datasets/imagenet-1k) (also referred to as ILSVRC 2012, a collection of 1.3 million images and 1,000 classes).
-- The segmentation model uses a [DeepLabV3](https://huggingface.co/papers/1706.05587) head. The available semantic segmentation checkpoints are pre-trained on [PASCAL VOC](http://host.robots.ox.ac.uk/pascal/VOC/).
-- As the name suggests MobileViT was designed to be performant and efficient on mobile phones. The TensorFlow versions of the MobileViT models are fully compatible with [TensorFlow Lite](https://www.tensorflow.org/lite).
+Click on the MobileViT models in the right sidebar for more examples of how to apply Model name to different vision tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]:
```suggestion
```

File: docs/source/en/model_doc/mobilevit.md [hunk 7]
```diff
@@ -1,141 +1,214 @@
...
 
-  You can use the following code to convert a MobileViT checkpoint (be it image classification or semantic segmentation) to generate a
-  TensorFlow Lite model:
+The example below demonstrates how to  [Convert Mobile ViT image classification checkpoint to a TensorFlow Lite Model] using the  [`TFMobileViTForImageClassification`] class:
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]: Include an example of image classification with `Pipeline` and `AutoModel` and not converting the checkpoint

File: docs/source/en/model_doc/mobilevit.md [hunk 8]
```diff
@@ -1,141 +1,214 @@
...
 
-## Resources
 
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with MobileViT.
 
-<PipelineTag pipeline="image-classification"/>
 
-- [`MobileViTForImageClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/image-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/image_classification.ipynb).
-- See also: [Image classification task guide](../tasks/image_classification)
+## Notes
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md:98]: Add the [Usage tips](https://huggingface.co/docs/transformers/v4.53.3/en/model_doc/mobilevit#usage-tips) as bullet points under `## Notes`.

File: docs/source/en/model_doc/mobilevit.md [hunk 9]
```diff
@@ -1,141 +1,214 @@
...
+- Does **not** operate on sequential data, it's purely designed for image tasks.
+- Feature maps are used directly instead of token embeddings.
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
 
-## MobileViTConfig
 
-[[autodoc]] MobileViTConfig
 
+## Resources
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]: You can remove all the `## Resources` here

File: docs/source/en/model_doc/mobilevit.md [hunk 10]
```diff
@@ -1,141 +1,214 @@
...
+  
 ## MobileViTFeatureExtractor
 
+  
+
 [[autodoc]] MobileViTFeatureExtractor
-    - __call__
-    - post_process_semantic_segmentation
+
+- __call__
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]: Don't need to make any changes to the API references here

**Review by stevhliu — COMMENTED** [review]
Good start! Please check the model card format again as its missing `Pipeline` and `AutoModel` examples!

File: docs/source/en/model_doc/mobilevit.md (hunk 3 above)
Comment (Shivamjan) [reply docs/source/en/model_doc/mobilevit.md]: Hey @stevhliu , I have uploaded the image and pinged you there, can you please check and merge it. I have also made the changes you suggested in the model card, let me know if everything looks good. I will add the image to the model card once you merge it. Thanks

File: docs/source/en/model_doc/mobilevit.md [hunk 11]
```diff
@@ -1,84 +1,124 @@
...
+```python
 
-This model was contributed by [matthijs](https://huggingface.co/Matthijs). The TensorFlow version of the model was contributed by [sayakpaul](https://huggingface.co/sayakpaul). The original code and weights can be found [here](https://github.com/apple/ml-cvnets).
+from transformers import pipeline
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md:50]: You can just run it on a single image rather than a dataset

```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="image-classification",
    model="apple/mobilevit-small",
    torch_dtype=torch.float16,
    device=0
)
pipeline("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg")
```

File: docs/source/en/model_doc/mobilevit.md [hunk 12]
```diff
@@ -1,84 +1,124 @@
...
+	print(f"Prediction: {preds}\n")
+```
+
+</hfoption>
+
+<hfoption id="AutoModel">
+
+```python
+
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md:68]:
```py
import torch
import requests
from PIL import Image
from transformers import AutoModelForImageClassification, AutoImageProcessor

image_processor = AutoImageProcessor.from_pretrained(
    "apple/mobilevit-small",
    use_fast=True,
)
model = AutoModelForImageClassification.from_pretrained(
    "apple/mobilevit-small",
)
url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
image = Image.open(requests.get(url, stream=True).raw)
inputs = image_processor(image, return_tensors="pt").to("cuda")

with torch.no_grad():
  logits = model(**inputs).logits
predicted_class_id = logits.argmax(dim=-1).item()

class_labels = model.config.id2label
predicted_class_label = class_labels[predicted_class_id]
print(f"The predicted class label is: {predicted_class_label}")
```

File: docs/source/en/model_doc/mobilevit.md [hunk 13]
```diff
@@ -1,84 +1,124 @@
...
 
-## Resources
 
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with MobileViT.
+## Notes
 
-<PipelineTag pipeline="image-classification"/>
+- Does **not** operate on sequential data, it's purely designed for image tasks.
+- Feature maps are used directly instead of token embeddings.
+- Use [`MobileViTImageProcessor`](https://huggingface.co/docs/transformers/main/en/model_doc/mobilevit#transformers.MobileViTImageProcessor) to preprocess images.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]:
```suggestion
- Use [`MobileViTImageProcessor`] to preprocess images.
```

File: docs/source/en/model_doc/mobilevit.md [hunk 14]
```diff
@@ -1,84 +1,124 @@
...
+- If using custom preprocessing, ensure that images are in **BGR** format (not RGB), as expected by the pretrained weights.
+- The **classification models** are pretrained on [**ImageNet-1k**](https://huggingface.co/datasets/imagenet-1k) (ILSVRC 2012).
+- The **segmentation models** use a [**DeepLabV3**](https://huggingface.co/papers/1706.05587) head and are pretrained on [**PASCAL VOC**](http://host.robots.ox.ac.uk/pascal/VOC/).
+- TensorFlow versions are compatible with **TensorFlow Lite**, making them ideal for edge/mobile deployment.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]:
```suggestion
- The classification models are pretrained on [ImageNet-1k](https://huggingface.co/datasets/imagenet-1k).
- The segmentation models use a [DeepLabV3](https://huggingface.co/papers/1706.05587) head and are pretrained on [PASCAL VOC](http://host.robots.ox.ac.uk/pascal/VOC/).
- TensorFlow versions are compatible with TensorFlow Lite, making them ideal for edge/mobile deployment.
```

**Review by stevhliu — APPROVED** [review]
Thanks, just a few more comments!

**Comment by Shivamjan** [comment]
@stevhliu I have made the suggested changes, do take a look!

File: docs/source/en/model_doc/mobilevit.md [hunk 15]
```diff
@@ -1,84 +1,116 @@
...
+image = Image.open(requests.get(url, stream=True).raw)
+inputs = image_processor(image, return_tensors="pt").to("cuda")
+
+with torch.no_grad():
+    logits = model(**inputs).logits
+predicted_class_id = logits.argmax(dim=-1).item()
+
+class_labels = model.config.id2label
+predicted_class_label = class_labels[predicted_class_id]
+print(f"The predicted calss label is:{predicted_class_label}")
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]:
```suggestion
print(f"The predicted class label is:{predicted_class_label}")
```

File: docs/source/en/model_doc/mobilevit.md [hunk 16]
```diff
@@ -1,84 +1,116 @@
...
-tflite_filename = model_ckpt.split("/")[-1] + ".tflite"
-with open(tflite_filename, "wb") as f:
-    f.write(tflite_model)
+image_processor = AutoImageProcessor.from_pretrained(
+							"apple/mobilevit-small",
+                            use_fast = True,
+							)
+model = MobileViTForImageClassification.from_pretrained(
+    								"apple/mobilevit-small"
+                                     )
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]: Keep the arguments to `from_pretrained` on the same line or don't indent them as far as that makes it hard to read in the docs

File: docs/source/en/model_doc/mobilevit.md [hunk 17]
```diff
@@ -1,84 +1,116 @@
...
 
-```py
-from transformers import TFMobileViTForImageClassification
-import tensorflow as tf
 
+classifier = pipeline(task = "image-classification",
+					  model = "apple/mobilevit-small",
+					  torch_dtype = torch.float16,
+                       device = 0
+                      )
```
Comment (stevhliu) [inline docs/source/en/model_doc/mobilevit.md]: No need to indent these all the way either as its hard to read in the docs

**Review by stevhliu — COMMENTED** [review]
Nice, just a few more formatting issues to take care of :)

**Comment by Shivamjan** [comment]
Take a look, I hope it looks good now.

**Comment by ArthurZucker** [comment]
Kudos this is the 20 000th commit! 🤗
